### PR TITLE
[DTRA]/Ahmad/DTRA-1147/GH Workflow Alerts

### PR DESCRIPTION
**Pull request by ahmadtaimoor-deriv**
Changes:
All the Risky content in these workflows is moved to environment variables and are accessed via it

-   ...

## Type of change

-   [x] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: .github/actions/versioning/action.yml [hunk 1]
```diff
@@ -8,5 +8,7 @@ runs:
   using: composite
   steps:
   - name: Tag build
-    run: echo "${{ inputs.target_branch }} $(date -u +'%Y-%m-%dT%H:%M:%SZ')" > dist/version.txt
+    env:
+    TARGET_BRANCH: ${{ inputs.target_branch }}    
+    run: echo "${TARGET_BRANCH} $(date -u +'%Y-%m-%dT%H:%M:%SZ')" > dist/version.txt
```
Comment (balakrishna-deriv) [inline .github/actions/versioning/action.yml]: format issue

File: .github/actions/notify_slack/action.yml [hunk 2]
```diff
@@ -19,22 +19,30 @@ runs:
...
+        SLACK_WEBHOOK_URL : ${{ inputs.SLACK_WEBHOOK_URL }}
     if: inputs.status == 'failure'
     run: |
           curl -X POST -H 'Content-type: application/json' \
           --data '{
-            "text": "${{ inputs.RELEASE_TYPE }} Release failed for smarttrader.deriv.com with version ${{ inputs.VERSION }}"
+            "text": "${RELEASE_TYPE} Release failed for smarttrader.deriv.com with version ${VERSION}"
           }' \
-          ${{ inputs.SLACK_WEBHOOK_URL }}
+          ${SLACK_WEBHOOK_URL}
```
Comment (balakrishna-deriv) [inline .github/actions/notify_slack/action.yml]: It should be 
```suggestion
          $SLACK_WEBHOOK_URL
```

File: .github/actions/versioning/action.yml [hunk 3]
```diff
@@ -2,11 +2,13 @@ name: versioning
...
     required: false
 runs:
   using: composite
   steps:
-  - name: Tag build
-    run: echo "${{ inputs.target_branch }} $(date -u +'%Y-%m-%dT%H:%M:%SZ')" > dist/version.txt
-    shell: bash
+    - name: Tag build
+      env:
+      TARGET_BRANCH: ${{ inputs.target_branch }}
```
Comment (balakrishna-deriv) [inline .github/actions/versioning/action.yml]: this line is not formatted

File: .github/workflows/release_production.yml [hunk 4]
```diff
@@ -3,6 +3,9 @@ on:
   push:
     tags:
       - production_*
+env:
+  RELEASE_TYPE: Production
+  APP_NAME: smarttrader-deriv-app
```
Comment (balakrishna-deriv) [inline .github/workflows/release_production.yml]: this should be moved to the step scope as this is just for one step.

File: .github/workflows/release_production.yml [hunk 5]
```diff
@@ -89,9 +89,10 @@ jobs:
...
+        env:
+          RELEASE_TYPE: Production
+          APP_NAME: smarttrader-deriv-app
+        uses: "deriv-com/shared-actions/.github/actions/send_slack_notification@master"
         with:
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
-          status: ${{ env.WORKFLOW_CONCLUSION }}
-          release_type: Production
-          version: ${{ steps.extract_version.outputs.RELEASE_VERSION }}
+          MESSAGE: "$RELEASE_TYPE Docker Publish and Kubernetes Deployment for $APP_NAME with version ${{ needs.build_test_and_publish.outputs.RELEASE_VERSION }} has Failed"
```
Comment (balakrishna-deriv) [inline .github/workflows/release_production.yml]: We should compose the slack message like here - https://github.com/binary-com/deriv-app/pull/14295/files#diff-93c7ed8fd6afff1de03995a214df3faa4dcd98168fed9759c

Can we check this?

image
17e5db6e3147a09R77-R89

File: .github/workflows/release_production.yml [hunk 6]
```diff
@@ -88,10 +95,16 @@ jobs:
       - name: Grab Version Name
         id: extract_version
         run: echo "RELEASE_VERSION=$(cat dist/version.txt)" >> $GITHUB_ENV
+      - name: Create Slack Message
+        id: create_slack_message
```
Comment (balakrishna-deriv) [inline .github/workflows/release_production.yml:103]: Can we add the RELEASE_TYPE and APP_NAME env here?

Comment (ahmadtaimoor-deriv) [reply .github/workflows/release_production.yml:103]: done

**Review by balakrishna-deriv — DISMISSED** [review]

**Review by ritik-deriv — APPROVED** [review]
LGTM

**Pull request merged**